### PR TITLE
Removing unused BC

### LINKED ISSUE #49: Multiple Ionic Species in BCs
If Zapdos is meant to be a general plasma simulation tool moving forward, we're going to need to allow multiple ionic species to be included in the model. Right now most relevant boundary conditions are hard-coded to allow a single ion species (example: _ip = 'Arp'_) and a single set of material properties associated with that one ion. This should be changed to allow the user to input a vector of ions. These are the boundary conditions that should be modified: 

- [x] FieldEmissionBC
- [x] HagelaarEnergyBC
- [x] LymberopoulosElectronBC
- [x] NeumannCircuitVoltageMoles_KV
- [x] NeumannCircuitVoltageNew
- [x] PenaltyCircuitPotential
~- [ ] PotentialDriftOutflowBC~
- [x] SakiyamaEnergySecondaryElectronBC
- [x] SakiyamaSecondaryElectronBC
- [x] SchottkyEmissionBC
- [x] SecondaryElectronEmissionBC

And of course this will need to allow for multiple secondary electron emission coefficients. SecondaryElectronEmissionBC can probably be left alone since it can be applied to each ion individually, but the formulation of many of these (like HagelaarEnergyBC) require a sum over all of the ion fluxes unless they are rewritten and a separate BC is included.

### PR DESCRIPTION
**Pull request by gsgall**
- Removes 'NeumannCircuitVoltageNew' since it is unregistered and untested
- This removal closes #49.

### CODE REVIEW

**Review by cticenhour — CHANGES_REQUESTED** [review]
Please perform a search for the object and remove it from the input files it appears in as well. I did this earlier, and they should be in the Schottky emission inputs, but commented out.

**Comment by gsgall** [comment]
> Please perform a search for the object and remove it from the input files it appears in as well. I did this earlier, and they should be in the Schottky emission inputs, but commented out.

I removed the comments where this was in there from 

- `tests/Schottky_emission/Example4/Input.i`
- `tests/Schottky_emission/Example4/Jac.i`
- `tests/Schottky_emission/Example4/Jac.i`

File: test/tests/1d_dc/NonlocalPotentialBCWithSchottky.i [hunk 1]
```diff
@@ -498,36 +498,6 @@ area = 5.02e-7 # Formerly 3.14e-6
 
 [BCs]
   ## Potential boundary conditions ##
```
Comment (cticenhour) [inline test/tests/1d_dc/NonlocalPotentialBCWithSchottky.i:500]:
```suggestion
  ## Potential boundary conditions ##
  #       [potential_left]
  #               type = NeumannCircuitVoltageMoles_KV
  #               variable = potential
  #               boundary = left
  #               function = potential_bc_func
  #               ions = Arp
  #               data_provider = data_provider
  #               electrons = em
  #               electron_energy = mean_en
  #               r = 0
  #               position_units = ${dom0Scale}
  #       []
  
```

`NeumannCircuitVoltageMoles_KV` isn't being removed here, so let's bring this half of the comments back.

File: test/tests/Schottky_emission/Example4/Input.i [hunk 2]
```diff
@@ -500,32 +500,6 @@ area = 5.02e-7 # Formerly 3.14e-6
 
 [BCs]
   ## Potential boundary conditions ##
```
Comment (cticenhour) [inline test/tests/Schottky_emission/Example4/Input.i:502]:
```suggestion
  ## Potential boundary conditions ##
  #       [potential_left]
  #               type = NeumannCircuitVoltageMoles_KV
  #               variable = potential
  #               boundary = left
  #               function = potential_bc_func
  #               ions = Arp
  #               data_provider = data_provider
  #               electrons = em
  #               electron_energy = mean_en
  #               r = 0
  #               position_units = ${dom0Scale}
  #       []
  
```

File: test/tests/Schottky_emission/Example4/Jac.i [hunk 3]
```diff
@@ -496,36 +496,6 @@ area = 5.02e-7 # Formerly 3.14e-6
 
 [BCs]
   ## Potential boundary conditions ##
```
Comment (cticenhour) [inline test/tests/Schottky_emission/Example4/Jac.i:498]:
```suggestion
  ## Potential boundary conditions ##
  #       [potential_left]
  #               type = NeumannCircuitVoltageMoles_KV
  #               variable = potential
  #               boundary = left
  #               function = potential_bc_func
  #               ions = Arp
  #               data_provider = data_provider
  #               electrons = em
  #               electron_energy = mean_en
  #               r = 0
  #               position_units = ${dom0Scale}
  #       []
  
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Thanks for removing the inputs blocks, but we should (for the moment) leave some of it in. See below.

**Comment by gsgall** [comment]
Just for my own knowledge is there a reason why we'd want to have extra input blocks commented out in test files. Is it just to show users other similar options?

**Comment by cticenhour** [comment]
I would normally not leave commented out blocks in, but I'm making an exception here because these inputs aren't fully tested - only syntax-checked. If they are ever fully tested (i.e., diffed), then we might need to know that an alternative currently-tested object was used for a particular case to help finalize the test.

One more thing that CI is saying we should do (because GitHub code suggestions can leave unintended whitespace). Please do what the precheck is telling us and fixup the last commit:

```
ERROR: The following files contain trailing whitespace after applying your patch:
	test/tests/1d_dc/NonlocalPotentialBCWithSchottky.i
	test/tests/Schottky_emission/Example4/Input.i
	test/tests/Schottky_emission/Example4/Jac.i

Run the "delete_trailing_whitespace.sh" script in your $MOOSE_DIR/scripts directory.
```

**Comment by gsgall** [comment]
that makes sense thank you for the explanation @cticenhour

**Review by cticenhour — APPROVED** [review]
